docs/grid-functionalities/FAQ.md

```diff
@@ -0,0 +1,53 @@
+#### index
+- Frequently asked questions
+  - [Merging grid options with applied defaults](#merging-grid-options-with-applied-defaults)
+
+### Description
+When working with the grid, you might want to Add / Update or Hightlight an item row from the Datagrid.
+
+**Note:** This is strictly a client side event, you still have to implement any backend change yourself.
+
+### Demo
+[Demo Page](https://ghiscoding.github.io/Angular-Slickgrid/#/additem) / [Demo Component](https://github.com/ghiscoding/angular-slickgrid/blob/master/src/app/examples/grid-additem.component.ts)
+
+## Frequently asked questions
+### Merging grid options with applied defaults
+When you pass gridOptions to the `angular-slickgrid` component, keep in mind that they get overloaded with the [Default Grid Options](https://github.com/ghiscoding/angular-slickgrid/blob/master/src/app/modules/angular-slickgrid/global-grid-options.ts). In contrast to what might be expected, this change won't overwrite your provided object.
+
+In cases, where depending on your data you might want to update the options (e.g. make columns readonly based on permissions) make sure to update your reference in the onAngularGridCreated event handler as shown below:
+
+#### View
+```html
+<angular-slickgrid
+     gridId="grid1"
+     [columnDefinitions]="columnDefinitions"
+     [gridOptions]="gridOptions"
+     [dataset]="dataset"
+     (onAngularGridCreated)="angularGridReady($event.detail)">
+</angular-slickgrid>
+```
+
+#### Component
+```typescript
+import { Component, OnInit} from '@angular/core';
+import { AngularGridInstance } from 'angular-slickgrid';
+
+export class GridBasicComponent implements OnInit {
+  columnDefinitions: Column[];
+  gridOptions: GridOption = {
+    // your initial settings
+  };
+  dataset: any[];
+
+  ngOnInit(): void {
+    this.columnDefinitions = [];
+  }
+
+  angularGridReady(angularGrid: AngularGridInstance) {
+    this.angularGrid = angularGrid;
+
+    // update your reference to make use of applied defaults
+    this.gridOptions = this.angularGrid.gridService._gridOptions as GridOption;
```

Comment:
is that intentional to reference `_gridOptions`?

Reply:
you meant instead of spreading it to a new instance? I dunno ... do you think this could result in any side-effects? Besides, it seems that the merging of gridOptions happens only once during `initialization`. So shouldn't be a big deal either way I'd guess

Reply:
I mean you can always get the current grid options from the grid instance, like `this.angularGrid.slickGrid.getOptions()` will get you the current grid options, instead of referencing `_gridOptions` which is protected and is meant to be used internally

Reply:
oh ok didnt notice that as its declared as public https://github.com/ghiscoding/slickgrid-universal/blob/211180b8c1f32250e6fc7a559baaa203154473e0/packages/common/src/services/grid.service.ts#L49

guess the underscore should have been the statement. Moreover I wasnt aware of the Slickgrid.getOptions() but will definitely change to use those

Reply:
ah right, I often forget to add the `protected` when I really should, so I'll change them later to really have the `protected` prop